controllers/component_image_controller.go

```diff
@@ -236,24 +243,33 @@ func (r *ComponentReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 					log.Error(nil, "Unknown error in the repository generation process", l.Audit, "true")
 					return ctrl.Result{}, r.reportError(ctx, component, "failed to generete image repository: unknown error")
 				}
-				log.Info(fmt.Sprintf("Created image repository %s for Component", repo.Name), l.Action, l.ActionAdd)
+				log.Info(fmt.Sprintf("Prepared image repository %s for Component", repo.Name), l.Action, l.ActionAdd)
 
 				imageURL := fmt.Sprintf("quay.io/%s/%s", r.QuayOrganization, repo.Name)
 
 				// Create secrets with the repository credentials
 				pushSecretName := component.Name
-				if err := r.EnsureRobotAccountSecret(ctx, component, pushRobotAccount, pushSecretName, imageURL); err != nil {
+				_, err := r.EnsureRobotAccountSecret(ctx, component, pushRobotAccount, pushSecretName, imageURL)
+				if err != nil {
 					return ctrl.Result{}, err
-				} else {
-					log.Info(fmt.Sprintf("Updated image registry push secret %s for Component", pushRobotAccount.Name), l.Action, l.ActionUpdate)
 				}
+				log.Info(fmt.Sprintf("Prepared image registry push secret %s for Component", pushRobotAccount.Name), l.Action, l.ActionUpdate)
 
 				pullSecretName := pushSecretName + "-pull"
-				if err := r.EnsureRobotAccountSecret(ctx, component, pullRobotAccount, pullSecretName, imageURL); err != nil {
+				pullSecretData, err := r.EnsureRobotAccountSecret(ctx, component, pullRobotAccount, pullSecretName, imageURL)
+				if err != nil {
 					return ctrl.Result{}, err
-				} else {
-					log.Info(fmt.Sprintf("Updated image registry pull secret %s for Component", pullRobotAccount.Name), l.Action, l.ActionUpdate)
 				}
+				log.Info(fmt.Sprintf("Prepared image registry pull secret %s for Component", pullRobotAccount.Name), l.Action, l.ActionUpdate)
+
+				// Propagate the pull secret into all environments
+				if err := r.EnsureComponentPullSecretRemoteSecret(ctx, component, pullSecretName); err != nil {
+					return ctrl.Result{}, err
+				}
+				if err := r.CreateRemoteSecretUploadSecret(ctx, component.Namespace, pullSecretName, pullSecretData); err != nil {
```

Comment:
do we need to keep the '-pull' secret in the namespace when it was uploaded already?